File: examples/python/bandcamp.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_bandcamp(url: str) -> dict:
    """
    Download track/album from Bandcamp
    
    Args:
        url: Bandcamp track or album URL
        
    Returns:
        dict: Download data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/bandcamp/download', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Bandcamp Track Downloaded Successfully!')
            print(f"Title: {data['data']['title']}")
            print(f"Artist: {data['data']['artist']}")
            print(f"Album: {data['data']['album']}")
            print(f"Genre: {data['data']['genre']}")
            print(f"Track Number: {data['data']['trackNumber']}")
            print(f"Download URL: {data['data']['downloadUrl']}")
            print(f"Duration: {data['data']['duration']} seconds")
            print(f"Release Date: {data['data']['releaseDate']}")
            print(f"Format: {data['data']['ext']}")
            
            return data['data']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_bandcamp_formats(url: str) -> list:
    """
    Get available formats for Bandcamp track
    
    Args:
        url: Bandcamp track URL
        
    Returns:
        list: Available formats or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/bandcamp/formats', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Available Formats:')
            for index, format_info in enumerate(data['formats'], 1):
                print(f'\nFormat {index}:')
                print(f"  ID: {format_info['formatId']}")
                print(f"  Extension: {format_info['ext']}")
                print(f"  Size: {format_info['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['formats']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_bandcamp_info(url: str) -> dict:
    """
    Get information about Bandcamp track/album
    
    Args:
        url: Bandcamp track or album URL
        
    Returns:
        dict: Track information or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/bandcamp/info', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Bandcamp Track Information:')
            print(f"Title: {data['info']['title']}")
            print(f"Artist: {data['info']['artist']}")
            print(f"Album: {data['info']['album']}")
            print(f"Genre: {data['info']['genre']}")
            print(f"Duration: {data['info']['duration']} seconds")
            print(f"Description: {data['info']['description']}")
            print(f"Release Date: {data['info']['releaseDate']}")
            
            return data['info']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    print('=== Bandcamp Download Example ===\n')
    
    # Example 1: Download Bandcamp track
    bandcamp_url = 'https://artist.bandcamp.com/track/example'
    print(f'Downloading from Bandcamp: {bandcamp_url}')
    download_bandcamp(bandcamp_url)
    
    print('\n=== Bandcamp Formats Example ===\n')
    
    # Example 2: Get available formats
    print(f'Getting formats for: {bandcamp_url}')
    get_bandcamp_formats(bandcamp_url)
    
    print('\n=== Bandcamp Info Example ===\n')
    
    # Example 3: Get track info
    print(f'Getting track info: {bandcamp_url}')
    get_bandcamp_info(bandcamp_url)

File: examples/nodejs/bandcamp.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download track/album from Bandcamp
 * @param {string} url - Bandcamp track or album URL
 */
async function downloadBandcamp(url) {
  try {
    const response = await axios.get(`${API_BASE}/bandcamp/download`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Bandcamp Track Downloaded Successfully!');
      console.log('Title:', response.data.data.title);
      console.log('Artist:', response.data.data.artist);
      console.log('Album:', response.data.data.album);
      console.log('Genre:', response.data.data.genre);
      console.log('Track Number:', response.data.data.trackNumber);
      console.log('Download URL:', response.data.data.downloadUrl);
      console.log('Duration:', response.data.data.duration, 'seconds');
      console.log('Release Date:', response.data.data.releaseDate);
      console.log('Format:', response.data.data.ext);
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get available formats for Bandcamp track
 * @param {string} url - Bandcamp track URL
 */
async function getBandcampFormats(url) {
  try {
    const response = await axios.get(`${API_BASE}/bandcamp/formats`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Available Formats:');
      response.data.formats.forEach((format, index) => {
        console.log(`\nFormat ${index + 1}:`);
        console.log('  ID:', format.formatId);
        console.log('  Extension:', format.ext);
        console.log('  Size:', (format.filesize / 1024 / 1024).toFixed(2), 'MB');
      });
      
      return response.data.formats;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get information about Bandcamp track/album
 * @param {string} url - Bandcamp track or album URL
 */
async function getBandcampInfo(url) {
  try {
    const response = await axios.get(`${API_BASE}/bandcamp/info`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Bandcamp Track Information:');
      console.log('Title:', response.data.info.title);
      console.log('Artist:', response.data.info.artist);
      console.log('Album:', response.data.info.album);
      console.log('Genre:', response.data.info.genre);
      console.log('Duration:', response.data.info.duration, 'seconds');
      console.log('Description:', response.data.info.description);
      console.log('Release Date:', response.data.info.releaseDate);
      
      return response.data.info;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  console.log('=== Bandcamp Download Example ===\n');
  
  // Example 1: Download Bandcamp track
  const bandcampUrl = 'https://artist.bandcamp.com/track/example';
  console.log('Downloading from Bandcamp:', bandcampUrl);
  await downloadBandcamp(bandcampUrl);
  
  console.log('\n=== Bandcamp Formats Example ===\n');
  
  // Example 2: Get available formats
  console.log('Getting formats for:', bandcampUrl);
  await getBandcampFormats(bandcampUrl);
  
  console.log('\n=== Bandcamp Info Example ===\n');
  
  // Example 3: Get track info
  console.log('Getting track info:', bandcampUrl);
  await getBandcampInfo(bandcampUrl);
})();

module.exports = {
  downloadBandcamp,
  getBandcampFormats,
  getBandcampInfo
};
